packages/buendia-update/data/usr/bin/buendia-update

```diff
@@ -54,11 +50,47 @@ if [ -n "$1" ]; then
 else
     packages=$(echo $(sed -e 's/#.*//' /usr/share/buendia/packages.list.d/*))
 fi
+
+# Attempt database migration first, if requested
+if echo $packages | grep buendia-server; then
+    echo "--- Attempting database migration"
+    apt-get -V -y --allow-unauthenticated install buendia-db-migrate
+
+    from_version=$(dpkg-query --showformat='${Version}' --show buendia-server)
+    to_version=$(dpkg-query --showformat='${Version}' --show buendia-db-migrate)
```

Comment:
The destination version is the version of the `buendia-db` package that is about to be installed.  Is there a possibility that `to_version` will be wrong because it's the latest version of `buendia-db-migrate`, rather than the latest version of `buendia-db`?

Reply:
(Can we think of a circumstance where they would be different?)

Reply:
Part of the problem with the question is that there's no easy way to know what version of a package "would" Debian install; it just installs it. Having separate `db-migrate` versus `db` packages helps this a bit because it allows the `db` package to ensure that the latest helper package can do its thing.

Reply:
We could parse the output of `apt-cache policy` but this doesn't seem to be documented nor intended for machine-readable consumption.

Reply:
Yeah, this doesn't seem likely to cause a problem in practice.  I think this is fine for now.